runtime/src/serde_snapshot.rs

```diff
@@ -442,51 +451,84 @@ where
                     )
                 })?;
 
+                // Remap the AppendVec ID to handle any duplicate IDs that may previously existed
+                // due to full snapshots and incremental snapshots generated from different nodes
+                let (new_append_vec_id, new_append_vec_path) = loop {
+                    let new_append_vec_id = next_append_vec_id.fetch_add(1, Ordering::Relaxed);
+                    let new_file_name = AppendVec::file_name(*slot, new_append_vec_id);
+                    let new_append_vec_path =
+                        append_vec_path.parent().unwrap().join(&new_file_name);
+
+                    // Make sure there's not another AppendVec already at this path.  If there is,
+                    // try again (and update the collision count), otherwise break out of the loop
+                    // with our new ID and path.
+                    let exists = std::fs::metadata(&new_append_vec_path).is_ok();
```

Comment:
Probably faster to just keep around a HashSet for all the appendvecs used in this slot, updated at the end of the loop here https://github.com/solana-labs/solana/blob/2e821371a588be0c9282b80b243d5a79138115df/runtime/src/serde_snapshot.rs#L480

Reply:
Does this still allow the same `append_vec_id` across different slots?

Reply:
AppendVec IDs will always be unique since the `next_append_vec_id` is Atomic. So no, the same AppendVec ID will _not_ occur on different slots.

As far as the HashSet, I like that idea. It becomes complicated for a few reasons, mostly due to the parallelization. I think it may be easier to talk about that on a call to see if there's a way around it. Is that OK?

Reply:
Makes sense! I misread the code, I thought we were only remapping ids for files that didn't exist. Oops!